### ISSUE TITLE
RequireNullSafeObjectOperatorSniff broken fix with ternary operator

### ISSUE THREAD

**Issue by schlndh** [slevomat/coding-standard#1569]
`RequireNullSafeObjectOperatorSniff` generates an incorrect fix for the following code:

```php
<?php

class TestDate
{
    public function __construct(private ?DateTimeImmutable $date)
    {
    }

    function format(?string $format): ?string
    {
        return $format !== null && $this->date !== null
            ? $this->date->format($format)
            : null;
    }
}
```
The generated fix is as follows:
```diff
     function format(?string $format): ?string
     {
-        return $format !== null && $this->date !== null
-            ? $this->date->format($format)
-            : null;
+        return $format !== null && $this->date?->format($format);
     }
 }
```
As you can see it completely changes the meaning of the code.

As a workaround I can put `$format !== null && $this->date !== null` into parentheses and then the issue is not detected at all. In this case the code could be simplified with nullsafe operator, but I'd certainly prefer to not detect the issue at all, instead of trying to fix it incorrectly.

Expected behavior:

Either ignore the issue, report it but don't offer a fix, or generate a fix as follows:

```diff
     function format(?string $format): ?string
     {
-        return $format !== null && $this->date !== null
-            ? $this->date->format($format)
+        return $format !== null
+            ? $this->date?->format($format)
             : null;
     }
 }
```

**Comment by kukulich** [comment]
Thank you for your report.

Fixed in https://github.com/slevomat/coding-standard/commit/7d9062c6df456b2ce8bf215fd0b3a9b9b66a3c9f
